### DISCUSSION TITLE
MAUI Push - Foreground Question

### DISCUSSION THREAD

**Discussion by jjohnnyl** [shinyorg/shiny#1272] [Q&A]
Trying to setup push notifications for my Maui app and following the example on the website:

using System.Reactive.Linq;
using Shiny;
using Shiny.Push;

public class PushForeground
{
    public void YourMethod()
    {
        var push = Host.Current.Services.GetService<IPushManager>(); // assign through DI, static, or ShinyHost.Resolve
        var disp = push
            .WhenReceived()
            .Where(x => x["newdata"] == "true")
            .SubscribeAsync(async data =>
            {
                // make you HTTP call here
            });
    }
}

But when i try this bit of code it says IPushManager (this is resolving fine) does not contain defintion for WhenReceived()?

What am I missing?

**Comment by aritchie** [comment]
The foreground method no longer exists.  The docs just haven't been updated

**Comment by jjohnnyl** [comment]
Ah thanks no wonder why I couldn't see in the code either, what's the best way to receive foreground notificaitons? The pushdelegate doesn't seem to pick up anything when the app is in the foreground(background works fine), also what happens with silent notifications?

**Reply by aritchie** [reply]
It isn't finished yet.  You can file PR's.  

You really don't want to use push notifications for foreground especially on ios since you only get so many in an hour.  Use signalr if you want realtime within your app
